### PR TITLE
Bugfix/extract personal detail validations to common

### PR DESCRIPTION
**Pull request by evbambly**
Extracted Personal Detail Validation logic to Common,
introducing common Errors, Validation functions and error translations.

In addition,
this solves the following [Trello Task](https://trello.com/c/KmftK5dK/490-coordinator-manual-booking-no-validation-on-input-in-donors-details)

### CODE REVIEW

**Comment by yaron1m** [comment]
Friendly ping

File: packages/common/src/LocaleUtils.ts [hunk 1]
```diff
@@ -72,3 +73,23 @@ export const getStatusTranslation = (
...
+      return "השם המלא ארוך מ-20 תווים";
+    case PersonalDetailsValidation.PHONE_INVALID:
+      return "מספר הטלפון אינו תקין";
+    case PersonalDetailsValidation.PHONE_HAS_NANS:
+      return "יש להזין ספרות בלבד";
+    case PersonalDetailsValidation.REQUIRED_FIELD:
+      return "שדה חובה";
+    case PersonalDetailsValidation.VALID_FIELD:
+    default:
+      return "";
```
[the remark this answers is not in the document]
Comment (evbambly) [reply packages/common/src/LocaleUtils.ts:93]: Done

File: packages/common/src/Validation.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  return PersonalDetailsValidation.VALID_FIELD;
+};
+
+export const ValidateLastName = (
+  lastName: string,
+  firstName: string
+): LastNameValidation => {
+  if (!lastName) return PersonalDetailsValidation.REQUIRED_FIELD;
+
+  if (typeof firstName === undefined) {
```
[the remark this answers is not in the document]
Comment (evbambly) [reply packages/common/src/Validation.ts:41]: I misread the caller component.
Done.

File: packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx [hunk 3]
```diff
@@ -26,52 +29,43 @@ export default function BookManualDonationScreen(
   props: BookManualDonationScreenProps
 ) {
   const [firstName, setFirstName] = useState("");
-  const [firstNameError, setFirstNameError] = useState("");
+  const [firstNameError, setFirstNameError] =
+    useState<Validation.FirstNameValidation>();
```
[the remark this answers is not in the document]
Comment (evbambly) [reply packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx:33]: Done

File: packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx [hunk 4]
```diff
@@ -26,52 +29,43 @@ export default function BookManualDonationScreen(
...
+  const setAndValidateFirstName = (nextFirstName: string): void => {
+    setFirstName(nextFirstName);
+    setFirstNameError(Validation.ValidateFirstName(nextFirstName));
+    // Validate long full name
+    if (
+      Validation.ValidateLastName(lastName, nextFirstName) ===
+        Validation.PersonalDetailsValidation.FULL_NAME_TOO_LONG ||
+      lastNameError === Validation.PersonalDetailsValidation.FULL_NAME_TOO_LONG
+    )
+      setLastNameError(Validation.ValidateLastName(lastName, nextFirstName));
```
[the remark this answers is not in the document]
Comment (evbambly) [reply packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx:53]: Done

File: packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx [hunk 5]
```diff
@@ -26,52 +29,43 @@ export default function BookManualDonationScreen(
...
-      valid = false;
-    } else if (lastName.length < 2) {
-      setLastNameError("הקלט אינו תקין");
-      valid = false;
-    }
-    if (!phone) {
-      setPhoneError("שדה חובה");
-      valid = false;
-    }
+  const setAndValidateFirstName = (nextFirstName: string): void => {
```
[the remark this answers is not in the document]
Comment (evbambly) [reply packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx:44]: The last name validation covers that scenario

File: packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx [hunk 6]
```diff
@@ -30,12 +30,18 @@ export default function BookManualDonationScreen(
 ) {
   const [firstName, setFirstName] = useState("");
   const [firstNameError, setFirstNameError] =
-    useState<Validation.FirstNameValidation>();
+    useState<Validation.FirstNameValidation>(
```
Comment (yaron1m) [inline packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx:33]: The `<type>` is now redundant (here and below)

File: packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx [hunk 7]
```diff
@@ -45,12 +51,16 @@ export default function BookManualDonationScreen(
...
+      nextFirstName
+    );
     if (
-      Validation.ValidateLastName(lastName, nextFirstName) ===
+      nextLastNameError ===
         Validation.PersonalDetailsValidation.FULL_NAME_TOO_LONG ||
       lastNameError === Validation.PersonalDetailsValidation.FULL_NAME_TOO_LONG
     )
-      setLastNameError(Validation.ValidateLastName(lastName, nextFirstName));
+      setLastNameError(nextLastNameError);
```
Comment (yaron1m) [inline packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx:63]: Please wrap `if` with ` {}`

**Review by yaron1m — APPROVED** [review]

File: packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx (hunk 6 above)
Comment (evbambly) [reply packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx:33]: Without the type, 
it wouldn't be explicitly FirstNameValidation,
but PersonalDetailsValidation

File: packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx (hunk 7 above)
Comment (evbambly) [reply packages/coordinator/src/screens/bookManualDonationScreen/BookManualDonationScreen.tsx:63]: Done

**Pull request merged**